src/modules/Resources/Files.ts

```diff
@@ -395,26 +429,27 @@ class Files extends TagoIOModule<GenericModuleParams> {
     parts: { ETag: string; PartNumber: number }[],
     options?: UploadOptions
   ) {
-    const { fieldId } = options || {};
+    const { fieldId: fieldID, isPublic } = options || {};
     const path =
       options?.dashboard && options?.widget ? `/data/files/${options.dashboard}/${options.widget}` : "/files";
 
     const partsOrdered = parts.sort((a, b) => a.PartNumber - b.PartNumber);
 
-    const _headers = { "Content-Type": "multipart/form-data" };
-
     const result = await this.doRequest<{ file: string }>({
       path,
       method: "POST",
       params: {
-        ...(options?.blueprint_devices?.length > 0 && { blueprint_devices: options.blueprint_devices }),
+        ...(options?.blueprint_devices?.length > 0 && {
+          blueprint_devices: options.blueprint_devices,
+        }),
       },
       body: {
         multipart_action: "end",
-        upload_id: uploadID,
         filename,
+        field_id: fieldID,
+        public: isPublic,
+        upload_id: uploadID,
```

Comment:
- Multipart `end` was left as **JSON** because API returns errors with `FormData`
- Current SDK/Admin in Production sends `end` action 

See screenshots below:

`end` multipart action (Production):
<img width="255" alt="image" src="https://github.com/user-attachments/assets/63dc2306-e023-431b-a957-ce174d5836d1" />

`upload` multipart action (Production):
<img width="289" alt="image" src="https://github.com/user-attachments/assets/26ecb528-629b-47b1-a0b0-92787d685081" />